File: sdk/python/examples/test_agent_slot_kimi.py
#!/usr/bin/env python3
"""
AgentSlot smoke test with Kimi model.

Verifies that the new AgentSlot class and spawn() method work correctly
with a real LLM (kimi-k2.5 via local proxy).
"""

import sys
import os

sys.path.insert(0, os.path.dirname(os.path.dirname(os.path.abspath(__file__))))

from a3s_code import Agent, Orchestrator, AgentSlot
from conftest import find_config

def main():
    print("=== AgentSlot + Kimi Test (Python) ===\n")

    config_path = find_config()
    print(f"✓ Config: {config_path}\n")

    print("Creating agent...")
    agent = Agent.create(config_path)
    print("✓ Agent created\n")

    print("Creating orchestrator (from_agent mode)...")
    orch = Orchestrator.create(agent)
    print("✓ Orchestrator created\n")

    # Test 1: spawn() via AgentSlot — standalone, no role
    print("Test 1: spawn() via AgentSlot (standalone, no role)...")
    slot = AgentSlot(
        agent_type="general",
        prompt="Reply with exactly: SLOT_OK",
        description="AgentSlot smoke test",
        permissive=True,
        max_steps=3,
    )
    print(f"  {slot!r}")
    handle = orch.spawn(slot)
    print(f"  ✓ Subagent spawned: {handle.id}")
    result = handle.wait()
    print(f"  ✓ Result: {result!r}\n")

    # Test 2: spawn() with a role field set (role is carried but spawn ignores it)
    print("Test 2: spawn() with role='worker' (standalone mode, role field present)...")
    slot_worker = AgentSlot(
        agent_type="general",
        prompt="Reply with exactly: WORKER_OK",
        description="Worker slot smoke test",
        role="worker",
        permissive=True,
        max_steps=3,
    )
    print(f"  {slot_worker!r}")
    handle2 = orch.spawn(slot_worker)
    print(f"  ✓ Subagent spawned: {handle2.id}")
    result2 = handle2.wait()
    print(f"  ✓ Result: {result2!r}\n")

    print("=== All AgentSlot tests passed ✓ ===")

if __name__ == "__main__":
    main()

File: sdk/node/examples/skills/test_agent_slot_kimi.ts
/**
 * AgentSlot smoke test with Kimi model (Node.js / TypeScript).
 *
 * Verifies that the new AgentSlot interface and spawn() method work correctly
 * with a real LLM (kimi-k2.5 via local proxy).
 */

import { Agent, Orchestrator, AgentSlot } from '@a3s-lab/code';
import * as path from 'path';
import * as fs from 'fs';
import { fileURLToPath } from 'url';

const __filename = fileURLToPath(import.meta.url);
const __dirname = path.dirname(__filename);

async function main() {
  console.log('=== AgentSlot + Kimi Test (Node.js) ===\n');

  const configPath = path.join(
    __dirname, '..', '..', '..', '..', '..', '.a3s', 'config.hcl'
  );
  if (!fs.existsSync(configPath)) {
    console.error(`✗ Config not found: ${configPath}`);
    process.exit(1);
  }
  console.log(`✓ Config: ${configPath}\n`);

  // Create agent
  console.log('Creating agent...');
  const agent = await Agent.create(configPath);
  console.log('✓ Agent created\n');

  // Orchestrator backed by a real agent
  console.log('Creating orchestrator (fromAgent mode)...');
  const orch = Orchestrator.create(agent);
  console.log('✓ Orchestrator created\n');

  // Test 1: spawn() via AgentSlot — standalone, no role
  console.log('Test 1: spawn() via AgentSlot (standalone, no role)...');
  const slot: AgentSlot = {
    agentType: 'general',
    prompt: 'Reply with exactly: SLOT_OK',
    description: 'AgentSlot smoke test',
    permissive: true,
    maxSteps: 3,
  };
  const handle = orch.spawn(slot);
  console.log(`  ✓ Subagent spawned: ${handle.id}`);
  const result = handle.wait();
  console.log(`  ✓ Result: ${JSON.stringify(result)}\n`);

  // Test 2: spawn() with role field present (standalone, role carried but ignored)
  console.log("Test 2: spawn() with role='worker' (standalone mode)...");
  const slotWorker: AgentSlot = {
    agentType: 'general',
    prompt: 'Reply with exactly: WORKER_OK',
    description: 'Worker slot smoke test',
    role: 'worker',
    permissive: true,
    maxSteps: 3,
  };
  const handle2 = orch.spawn(slotWorker);
  console.log(`  ✓ Subagent spawned: ${handle2.id}`);
  const result2 = handle2.wait();
  console.log(`  ✓ Result: ${JSON.stringify(result2)}\n`);

  console.log('=== All AgentSlot tests passed ✓ ===');
}

main().catch((e) => {
  console.error('Test failed:', e);
  process.exit(1);
});
